Add aws out of tree cloud provider install/upgrade docs (#844)
rancher/rancher-docs: docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-cloud-providers/amazon.md

:::note Important:

> In Kubernetes 1.27 and later, you must use an out-of-tree AWS cloud provider. In-tree cloud providers have been deprecated, and the amazon cloud provider has been removed completely, and won't work after an upgrade to Kubernetes 1.27. The steps listed below are still required to set up an Amazon cloud provider. You can [set up an out-of-tree cloud provider for RKE1](#using-the-out-of-tree-aws-cloud-provider-for-rke1) after creating an IAM role and configuring the ClusterID.

Reviewer: ```suggestion
In Kubernetes 1.27 and later, you must use an out-of-tree AWS cloud provider. In-tree cloud providers have been deprecated, and the Amazon cloud provider has been removed completely, and won't work after an upgrade to Kubernetes 1.27. The steps listed below are still required to set up an Amazon cloud provider. You can [set up an out-of-tree cloud provider for RKE](#using-the-out-of-tree-aws-cloud-provider-for-rke1) after creating an IAM role and configuring the ClusterID.
```

:::

### Using Amazon Elastic Container Registry (ECR)

The kubelet component has the ability to automatically obtain ECR credentials, when the IAM profile mentioned in [Create an IAM Role and attach to the instances](#1-create-an-iam-role-and-attach-to-the-instances) is attached to the instance(s). When using a Kubernetes version older than v1.15.0, the Amazon cloud provider needs be configured in the cluster. Starting with Kubernetes version v1.15.0, the kubelet can obtain ECR credentials without having the Amazon cloud provider configured in the cluster.

### Using the Out-of-Tree AWS Cloud Provider for RKE1

1. [Node name conventions and other prerequisities ](https://cloud-provider-aws.sigs.k8s.io/prerequisites/) must be followed so that the cloud provider can find the instance. Rancher provisioned clusters don't support configuring `providerID`.

> When IP-based naming is used, the nodes must be named after the instance followed by the regional domain name (`ip-xxx-xxx-xxx-xxx.ec2.<region>.internal`). If you have a custom domain name set in the DHCP options, you must set `--hostname-override` on `kube-proxy` and `kubelet` to match this naming convention.

> When creating a [custom cluster](../../../../pages-for-subheaders/use-existing-nodes.md), you must add [`--node-name`](../../reference-guides/cluster-configuration/rancher-server-configuration/use-existing-nodes/rancher-agent-options) to the `docker run` node registration command to set `hostname-override` -- for example, `"$(hostname -f)"`). This can be done manually or by using **Show Advanced Options** in the Rancher UI to add **Node Name**.

Reviewer: @jakefhyde This line is still a problem. I returned to the Rancher UI hoping I would find something like these instructions but I can't confirm where users should go.

> This can be done manually or by using **Show Advanced Options** in the Rancher UI to add **Node Name**.

> When creating a [custom cluster](../../../../pages-for-subheaders/use-existing-nodes.md), you must add [`--node-name`](../../reference-guides/cluster-configuration/rancher-server-configuration/use-existing-nodes/rancher-agent-options) to the `docker run` node registration command to set `hostname-override` -- for example, `"$(hostname -f)"`). This can be done manually or by using **Show Advanced Options** in the Rancher UI to add **Node Name**.

Author: ![image](https://github.com/rancher/rancher-docs/assets/33796120/d1cc56c7-6a74-4fe3-ae70-5d5cd33920b1)
![image](https://github.com/rancher/rancher-docs/assets/33796120/cac8ad6c-0822-4df0-bcb2-6fd713c73a11)

> 2. Select **External (out-of-tree)** as the cloud provider. This sets `--cloud-provider=external` for Kubernetes components. This can also be done manually, by setting it in `cluster.yml`.

Reviewer 2: @martyav Which drop down do you refer here? Since I pushed changes earlier I think the block this is referring to has changed.

```
rancher_kubernetes_engine_config:
  cloud_provider:
    name: external
```

> 3. Install the AWS cloud controller manager after the cluster finishes provisioning. Note that the cluster isn't successfully provisioned and nodes are still in an `uninitialized` state until you deploy the cloud controller manager. This can be done via Helm charts in UI or manually.

Reviewer: This still needs to be explicated and I didn't seem to find anything pre-existing in the docs (unless I misunderstood something...)

> 3. Install the AWS cloud controller manager after the cluster finishes provisioning. Note that the cluster isn't successfully provisioned and nodes are still in an `uninitialized` state until you deploy the cloud controller manager. This can be done via Helm charts in UI or manually.

Reviewer 2: There's a `Helm charts installation` section below this, I'll link it so it's explicit

**Key** = `kubernetes.io/cluster/CLUSTERID` **Value** = `shared`.

> :::note

Reviewer 2: It shows up as this when loading docs with `yarn start` 

<img width="790" alt="Screenshot 2023-10-22 at 2 26 03 PM" src="https://github.com/rancher/rancher-docs/assets/16870565/2367915c-bc3d-416c-8e3d-7dcb04da5a43">

:::note

The upstream documentation for the AWS cloud controller manager can be found [here](https://kubernetes.github.io/cloud-provider-aws).

:::

### Helm Chart Installation

1. Click **☰**, then select the name of the cluster from the left navigation.

2. Select **Apps** > **Repositories**.

3. Click the **Create** button**.

4. Enter `https://kubernetes.github.io/cloud-provider-aws` in the **Index URL** field.

5. Select **Apps** > **Charts** from the left navigation and install **aws-cloud-controller-manager**.

6. Select the namespace, `kube-system`, and enable **Customize Helm options before install**.

7. Rancher-provisioned RKE nodes are tainted `node-role.kubernetes.io/controlplane` so you must update tolerations and the nodeSelector:

```
tolerations:
  - effect: NoSchedule
    key: node.cloudprovider.kubernetes.io/uninitialized
    value: 'true'
  - effect: NoSchedule
    value: 'true'
    key: node-role.kubernetes.io/controlplane

```

:::note 
	

> There's currently [a known issue](https://github.com/rancher/dashboard/issues/9249) which doesn't allow nodeSelector to be updated from the Rancher UI.  Continue installing the chart and then edit the daemonset manually to set the `nodeSelector`:

Reviewer 2: Since this is only fixed for Rancher v2.8, adding that note but keeping the msg for people trying to migrate on 2.7 using k8s 1.26

```
nodeSelector:
  node-role.kubernetes.io/controlplane: 'true'
```

:::

8. Add `get` to `clusterRoleRules`. This allows the cloud controller manager to get service accounts upon startup.

```
  - apiGroups:
      - ''
    resources:
      - serviceaccounts
    verbs:
      - create
      - get
```

9. Update container arguments:
```
  - '--use-service-account-credentials=true'
  - '--configure-cloud-routes=false'
```   

> 10. Install the chart and confirm that daemonset `aws-cloud-controller-manager` deploys successfully.

Reviewer 3: can you add a note that, if using the workaround from step 7, then you should manually go and edit the daemonset at this step in order for it to deploy

  Confirm [migration configuration values](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#default-configuration) and special case around [migrating IPAM controllers](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#node-ipam-controller-migration).

1. Update cluster config to enable leader migration

```yaml
spec:
  rkeConfig:
    machineSelectorConfig:
      - config:
          kube-controller-manager-arg:
            - enable-leader-migration=true
```

Note that cloud provider is still `aws` at this step.

```yaml
spec:
  rkeConfig:
    machineGlobalConfig:
      cloud-provider-name: aws
```

2. Cordon control plane nodes so aws cloud controller pods run on nodes onlyafter upgrading to external cloud provider.

3. To install AWS cloud controller manager with leader migration enabled, follow Steps 1-3 for deploying cloud controller manager chart.

Update container args to enable leader migration,
```
- '--enable-leader-migration=true' 
```

4. Install chart and confirm daemonset `aws-cloud-controller-manager` deploys successfully.

> 5. Update provisioning cluster to change cloud provider and remove leader migration args from kube-controller.

Reviewer 2: RKE1 uncordons nodes by default after upgrade, so this step isn't necessary. You can confirm this by checking the aws cloud controller manager pods start deploying on nodes as controlplane nodes upgrade one by one.

    name: external-aws
    useInstanceMetadataHostname: true
```

You must not enable `useInstanceMetadataHostname` when setting custom values for `hostname-override` for custom clusters. When creating a [custom cluster](../../../../pages-for-subheaders/use-existing-nodes.md), you can add [`--node-name`](../../reference-guides/cluster-configuration/rancher-server-configuration/use-existing-nodes/rancher-agent-options) to the `docker run` node registration command to set `hostname-override` - for example, `"$(hostname -f)"`). This can be done manually or by using **Show Advanced Options** in the Rancher UI to add **Node Name**.

2. Select cloud provider. 

Selecting **External Amazon (out-of-tree)** will set `--cloud-provider=external` and allow enabling `useInstanceMetadataHostname`. You must enable `useInstanceMetadataHostname` if not providing custom node name via `--node-name`. As mentioned in step 1, enabling `useInstanceMetadataHostname` will query ec2 metadata service and set `/hostname` as `hostname-override` for `kubelet` and `kube-proxy`:

```
rancher_kubernetes_engine_config:
  cloud_provider:
    name: external-aws
    useInstanceMetadataHostname: true/false
```

> Selecting **External (out-of-tree)** will set `--cloud-provider=external` for Kubernetes components. This can also be done manually, by setting it in `cluster.yml`.

Reviewer 3: can it be clarified why you would use this option as opposed to the new external-aws option?

Reviewer 2: I removed this block since it's no longer needed

```   

8. Add `get` to `verbs` for `serviceaccounts` resources in `clusterRoleRules`. This allows the cloud controller manager to get service accounts upon startup.

```
  - apiGroups:
      - ''
    resources:
      - serviceaccounts
    verbs:
      - create
      - get
```

9. Rancher-provisioned RKE nodes are tainted `node-role.kubernetes.io/controlplane` so you must update tolerations and the nodeSelector:

```
tolerations:
  - effect: NoSchedule
    key: node.cloudprovider.kubernetes.io/uninitialized
    value: 'true'
  - effect: NoSchedule
    value: 'true'
    key: node-role.kubernetes.io/controlplane

```

```
nodeSelector:
  node-role.kubernetes.io/controlplane: 'true'
```

:::note for Rancher `<v2.8.0`:
	
There's currently [a known issue](https://github.com/rancher/dashboard/issues/9249) which doesn't allow nodeSelector to be updated from the Rancher UI.  Continue installing the chart and then edit the daemonset manually to set the `nodeSelector`. 

:::

10. Install the chart and confirm that daemonset `aws-cloud-controller-manager` is running - verify `aws-cloud-controller-manager` pods are running in target namespace (`kube-system` unless modified in step 6). 

```

> 

Reviewer 3: there's an empty code block here

```

### Migrating to the Out-of-Tree AWS Cloud Provider for RKE1

In order to upgrade existing cluster with in-tree cloud provider to AWS cloud controller manager, you can run stop kube controller manager and install AWS cloud controller manager in many ways. Refer to [External cloud controller manager](https://cloud-provider-aws.sigs.k8s.io/getting_started/) for details.

When downtime is acceptable, you can switch to external cloud provider which removes in-tree components and then deploy charts to install AWS cloud controller manager as explained in [using out of tree cloud provider](#using-out-of-tree-aws-cloud-provider).

> When control plane cannot tolerate downtime, leader migration must be enabled to facilitate a smooth migration from the controllers in the kube controller manager to their counterparts in the cloud controller manager. Refer to [Using leader migration](https://cloud-provider-aws.sigs.k8s.io/getting_started/) for more details.

Reviewer: ```suggestion
If your setup can't tolerate any control plane downtime, you must enable leader migration. This facilitates a smooth transition from the controllers in the kube controller manager to their counterparts in the cloud controller manager. Refer to the official AWS documentation on [Using leader migration](https://cloud-provider-aws.sigs.k8s.io/getting_started/) for more details.
```

**Important**

> - [Cloud controller migration docs](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#before-you-begin) mentions that it is possible to migrate with the same Kubernetes version, but assumes that migration is part of Kubernetes upgrade.

Reviewer: ```suggestion
- The Kubernetes [cloud controller migration documentation](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#before-you-begin) mentions that it is possible to migrate with the same Kubernetes version, but assumes that migration is part of a  Kubernetes upgrade.
```

> - Refer [Migrate to use cloud controller manager](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/) to confirm if any customizations are required before migrating.

Reviewer: ```suggestion
- Refer to the Kubernetes documentation on [migrating to use the cloud controller manager](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/) to see if you need to customize your setup before migrating.
```

I wasn't sure what "any customizations" referred to so I put "setup" as a generic placeholder.

>   Confirm [migration configuration values](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#default-configuration) and special case around [migrating IPAM controllers](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#node-ipam-controller-migration).

Reviewer: ```suggestion
  Confirm your [migration configuration values](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#default-configuration). If your cloud provider provides an implementation of the Node IPAM controller,  you also need to [migrate the IPAM controller](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#node-ipam-controller-migration).
```

> 1. Update cluster config to enable leader migration in `cluster.yml`

Reviewer: ```suggestion
1. Update the cluster config to enable leader migration in `cluster.yml`
```

```
services:
  kube-controller:
    extra_args:
      enable-leader-migration: "true"
```

> Note that cloud provider is still `aws` at this step.

Reviewer: ```suggestion

Note that the cloud provider is still `aws` at this step.

```

```
cloud_provider:
  name: aws
```

> 2. Cordon control plane nodes so aws cloud controller pods run on nodes only after upgrading to external cloud provider.

Reviewer: ```suggestion
2. Cordon the control plane nodes, so that AWS cloud controller pods run on nodes only after upgrading to the external cloud provider.
```

```
kubectl cordon -l "node-role.kubernetes.io/controlplane=true"
```

> 3. The steps to install AWS cloud controller manager for migration are same as fresh installation, but you must enable leader migration. 

Reviewer: ```suggestion
3. To install the AWS cloud controller manager, enable leader migration and follow the same steps as when installing AWS on a new cluster. 
```

> Add the following to the container arguments in step 7 when following the [steps to install chart](#helm-chart-installation):

Reviewer: ```suggestion
To enable leader migration, add the following to the container arguments in step 7 while following the [steps to install the chart](#helm-chart-installation):
```

```
- '--enable-leader-migration=true' 
```

> 4. Confirm the chart is installed but the new pods aren't running yet due to cordoned controlplane nodes. After updating the cluster in thenext step, rke will uncordon each node after upgrading and `aws-controller-manager` pods will be scheduled.

Reviewer 3: `in thenext step` small typo

> 4. Confirm the chart is installed but the new pods aren't running yet due to cordoned controlplane nodes. After updating the cluster in thenext step, rke will uncordon each node after upgrading and `aws-controller-manager` pods will be scheduled.

Reviewer: ```suggestion
4. Confirm the chart is installed but the new pods aren't running yet due to cordoned controlplane nodes. After updating the cluster in the next step, RKE uncordons each node after upgrading and schedules `aws-controller-manager` pods.
```

Reviewer 2: @martyav I am a bit confused for this change, the uncordoning of node will happen only after updating cluster and upgrading nodes so shouldn't it be `will uncordon`?

Reviewer: Maybe a little overzealous in changing tense. We can keep in "will" if that helps with clarity.

> 4. Confirm the chart is installed but the new pods aren't running yet due to cordoned controlplane nodes. After updating the cluster in the next step, rke will uncordon each node after upgrading and `aws-controller-manager` pods will be scheduled.

Reviewer: ```suggestion
4. Confirm that the chart is installed but the new pods aren't running yet due to cordoned controlplane nodes. After updating the cluster in the next step, RKE will uncordon each node after upgrading and `aws-controller-manager` pods will be scheduled.
```

> 5. Update cluster.yml to change cloud provider and remove leader migration args from kube-controller.

Reviewer: ```suggestion
5. Update cluster.yml to change the cloud provider and remove the leader migration arguments from the kube-controller.
```

> Selecting **External Amazon (out-of-tree)** will set `--cloud-provider=external` and allow enabling `useInstanceMetadataHostname`. You must enable `useInstanceMetadataHostname` for node-driver clusters and for custom clusters if not providing custom node name via `--node-name`. Enabling `useInstanceMetadataHostname` will query ec2 metadata service and set `/hostname` as `hostname-override` for `kubelet` and `kube-proxy`:

Reviewer 3: `if not providing custom node name` -> since this is basically an upgrade case where nodes are already registered. Would the user add `useInstanceMetadataHostname` by default then for custom clusters too? 

I have the following use cases that would be great to have clarity on, what should the user do if:
* they already have nodes registered without using --node-name
* they have nodes registered and did use --node-name

if we instruct them to enable it, what happens if:
* they plan on adding more nodes later on and use --node-name override?

Author: @kinarashah Need clarification on this

Reviewer 2: Sorry @jakefhyde, I missed the tag notification.  

@slickwarren The main rule here is that node-name should match what the cloud provider expects. If not, users would need to use `useInstanceMetadataHostname` to let rancher query the ec2 service. So 
- they already have nodes registered without using --node-name => use `useInstanceMetadataHostname` 
- they have nodes registered and did use --node-name => if node-name matches what the cloud provider expects, they're good. If not, use `useInstanceMetadataHostname`  

- if we instruct them to enable it, what happens if: they plan on adding more nodes later on and use --node-name override? => this is tricky, but since the logic is at cluster level, they'd need to continue using whatever approach they used for existing nodes unless the existing nodes are reregistered the same way as new nodes

```
rancher_kubernetes_engine_config:
  cloud_provider:
    name: external-aws
    useInstanceMetadataHostname: true/false
```

**Remove** `enable-leader-migration` from:
```
services:
  kube-controller:
    extra_args:
      enable-leader-migration: "true"
```

> 6. If upgrading Kubernetes version, set Kubernetes version as well.

Reviewer: ```suggestion
6. If  you're upgrading the cluster's Kubernetes version, set the Kubernetes version as well.
```

> 7. Update the cluster. There should be new `aws-cloud-controller-manager` pods running in the cluster.  

Reviewer: ```suggestion
7. Update the cluster. The `aws-cloud-controller-manager` pods should now be running.  
```

> 8. Optionally after the upgrade, AWS cloud controller manager can be updated to disable leader migration. Upgrade the chart and remove following section from container arguments:

Reviewer: ```suggestion
8. (Optional) After the upgrade, you can update the AWS cloud controller manager to disable leader migration. Upgrade the chart and remove the following section from the container arguments:
```

    resources:
      - nodes
    verbs:
      - '*'
  - apiGroups:
      - ""
    resources:
      - nodes/status
    verbs:
      - patch
  - apiGroups:
      - ""
    resources:
      - services
    verbs:
      - list
      - patch
      - update
      - watch
  - apiGroups:
      - ""
    resources:
      - services/status
    verbs:
      - list
      - patch
      - update
      - watch
  - apiGroups:
     - ''
    resources:
      - serviceaccounts
    verbs:
    - create
    - get
  - apiGroups:
      - ""
    resources:
      - persistentvolumes
    verbs:
      - get
      - list
      - update
      - watch
  - apiGroups:
      - ""
    resources:
      - endpoints
    verbs:
      - create
      - get
      - list
      - watch
      - update
  - apiGroups:
      - coordination.k8s.io
    resources:
      - leases
    verbs:
      - create
      - get
      - list
      - watch
      - update
  - apiGroups:
      - ""
    resources:
      - serviceaccounts/token
    verbs:
      - create
```

3. Install the helm chart:

```shell
helm upgrade --install aws-cloud-controller-manager aws-cloud-controller-manager/aws-cloud-controller-manager --values values.yaml
```

> Verify that the helm chart installed successfully:

Reviewer 2: Default chart has an extra node selector which needs to be removed, otherwise it won't schedule on rke1 nodes: 

4. If present, edit daemonset to remove the default node selector `node-role.kubernetes.io/control-plane: ""`:

```shell
kubectl edit daemonset aws-cloud-controller-manager -n kube-system
```

```

:::tip
You can also disable leader migration after step 7. Upgrade the chart and remove the following section from the container arguments:

```yaml
- --enable-leader-migration=true 
```

:::

6. If  you're upgrading the cluster's Kubernetes version, set the Kubernetes version as well.

7. Update the cluster. The `aws-cloud-controller-manager` pods should now be running.

#### Migrating to the Out-of-Tree AWS Cloud Provider for RKE2

1. Update the cluster config to enable leader migration:

```yaml
spec:
  rkeConfig:
    machineSelectorConfig:
      - config:
          kube-controller-manager-arg:
            - enable-leader-migration
        machineLabelSelector:
          matchExpressions:
            - key: rke.cattle.io/control-plane-role
              operator: In
              values:
                - 'true'
```

Note that the cloud provider is still `aws` at this step:

```yaml
spec:
  rkeConfig:
    machineGlobalConfig:
      cloud-provider-name: aws
```

2. Cordon control plane nodes so that AWS cloud controller pods run on nodes only after upgrading to the external cloud provider:

```shell
kubectl cordon -l "node-role.kubernetes.io/controlplane=true"
```

> 3. To install AWS cloud controller manager with leader migration enabled, follow Steps 1-3 for [deploying cloud controller manager chart](#using-out-of-tree-aws-cloud-provider-for-rke2)

Reviewer: ```suggestion
3. To install the AWS cloud controller manager with leader migration enabled, follow Steps 1-3 for [deploying the cloud controller manager chart](#using-out-of-tree-aws-cloud-provider-for-rke2)
```

```yaml
rancher_kubernetes_engine_config:
  cloud_provider:
    name: external-aws
    useInstanceMetadataHostname: true/false
```

**Remove** `enable-leader-migration` from:

```yaml
services:
  kube-controller:
    extra_args:
      enable-leader-migration: "true"
```

6. If  you're upgrading the cluster's Kubernetes version, set the Kubernetes version as well.

7. Update the cluster. The `aws-cloud-controller-manager` pods should now be running.

8. (Optional) After the upgrade, leader migration is no longer required due to only one cloud-controller-manager and can be removed. Upgrade the chart and remove the following section from the container arguments:

```shell
- --enable-leader-migration=true 
```

Verify the cloud controller manager update was successfully rolled out with the following command:

```shell
kubectl rollout status daemonset -n kube-system aws-cloud-controller-manager
```

> ### Using Out-of-tree AWS Cloud Provider for RKE2

Reviewer 4: ```suggestion
### Using the Out-of-tree AWS Cloud Provider for RKE2
```

Reviewer: I was reviewing the document again in order to work towards the reorganization of the "Set up Cloud Providers" section proposed, and there's a very large duplicated section from 427 to 510. I'm currently editing the file to fix the issue

> 1. [Node name conventions and other prerequisites](https://cloud-provider-aws.sigs.k8s.io/prerequisites/) must be followed for cloud provider to find the instance correctly.

Reviewer: ```suggestion
1. [Node name conventions and other prerequisities ](https://cloud-provider-aws.sigs.k8s.io/prerequisites/) must be followed so that the cloud provider can find the instance.
```

```

Note that the cloud provider is still `aws` at this step:

```yaml
spec:
  rkeConfig:
    machineGlobalConfig:
      cloud-provider-name: aws
```

2. Cordon control plane nodes so that AWS cloud controller pods run on nodes only after upgrading to the external cloud provider:

```bash
kubectl cordon -l "node-role.kubernetes.io/controlplane=true"

3. To install AWS cloud controller manager with leader migration enabled, follow Steps 1-3 for [deploying cloud controller manager chart](#using-out-of-tree-aws-cloud-provider-for-rke2)
From Kubernetes 1.22 onwards, the kube-controller-manager will utilize a default configuration which will satisfy the controller-to-manager migration. 
Update container args of the `aws-cloud-controller-manager` under `spec.rkeConfig.additionalManifest` to enable leader migration:
```

3. To install the AWS cloud controller manager with leader migration enabled, follow steps 1-3 for [deploying a cloud controller manager chart](#using-the-out-of-tree-aws-cloud-provider-for-rke2). In step 3 of the fresh install steps, add the following container arg to the `additionalManifest` to enable leader migration:

```bash
- '--enable-leader-migration=true' 

> ```

Reviewer: Duplicate step 3 -- which to keep?
